src/backend/core/authentication/backends.py

```diff
@@ -101,13 +139,40 @@ def create_user(self, claims):
             raise SuspiciousOperation(
                 _("Claims contained no recognizable user identification")
             )
+        email = claims.get("email")
+        name = claims.get("name")
 
-        return self.UserModel.objects.create(
+        # Extract or create the organization from the data
+        domain = get_domain_from_email(email)
```

Comment:
It was not DRY on purpose, because I want the code for existing user to be deleted once this is deployed and all existing user have an organization. Not be DRY will allow easy cleanup (and after the cleanup only this part will remain).

Reply:
Maybe create an issue for this?

Reply:
Issue added https://github.com/numerique-gouv/people/issues/504